deploy-agent/setup.py

```diff
@@ -30,6 +30,8 @@
     "gevent==20.12.0; python_version >= '3.8'",
     "lockfile==0.10.2",
     "boto>=2.39.0",
+    "boto3>=1.12.6",
```

Comment:
Why not the latest version? 
1.34.36

Reply:
https://deps.dev/pypi/boto3/1.34.36/versions

Reply:
Given that we currently use requests==2.20.0, we need urllib3<1.25, >=1.21.1. The challenge here is that we must identify the precise boto3 and botocore versions which can adequately support urllib3 within the range of <1.25 and >=1.21.1.

Reply:
Why not upgrade requests if that's the case?

Reply:
The latest requests version is 2.31.0. And because it's a minor version change it shouldn't have breaking changes. Can we use that one?

Reply:
The principal aim of this pull request is to incorporate and utilize boto3. I'd prefer not to meddle with the current version of request. If you're keen to streamline the versioning aspect, you're more than welcome to invest some time enhancing and upgrading the versions. Thank you.